src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java

```diff
@@ -202,6 +202,27 @@ public UserTenant update(UUID consortiumId, UserTenant primary) {
     return new UserTenant();
   }
 
+  @Override
+  public void updateNames(UUID userId) {
+    List<UserTenantEntity> userTenantEntities = userTenantRepository.getByUserIdAndIsPrimaryFalse(userId);
+    if (CollectionUtils.isNotEmpty(userTenantEntities)) {
+      User user = userService.getById(userId);
+      String firstName = user.getPersonal().getFirstName();
+      String lastName = user.getPersonal().getLastName();
+      List<String> tenantIds = userTenantEntities.stream().map(userTenantEntity -> userTenantEntity.getTenant().getId()).toList();
+      log.info("Updating shadow users in all tenants exist in consortia for the user: {}", userId);
+      tenantIds.forEach(tenantId -> {
+        try (var ignored = new FolioExecutionContextSetter(prepareContextForTenant(tenantId, folioModuleMetadata, folioExecutionContext))) {
+          User shadowUser = userService.getById(userId);
```

Comment:
We can get real user only once before cycle started, because real user would be the same for all shadow users